commit f6c829ad141daab1a6ea5740bf4dde49ad03da31
Author: Satish Balay <balay@mcs.anl.gov>
Date:   Sun Oct 30 12:07:54 2022 -0500

    configure: bison is not needed for --with-ptscotch. Yet configure gives error (due to self.lookforbydefault=1):
    
    Package BISON needed by PTScotch failed to configure.
    
      File "/build/petsc-3.18.1+dfsg1/config/BuildSystem/config/package.py", line 914, in checkDependencies
        raise RuntimeError('Package '+package.PACKAGE+' needed by '+self.name+' failed to configure.\nMail configure.log to petsc-maint@mcs.anl.gov.')

diff --git a/config/BuildSystem/config/package.py b/config/BuildSystem/config/package.py
index c9437a17c5e..04fcbb862db 100644
--- a/config/BuildSystem/config/package.py
+++ b/config/BuildSystem/config/package.py
@@ -910,7 +910,7 @@ To use currently downloaded (local) git snapshot - use: --download-'+self.packag
       if not hasattr(package, 'found'):
         raise RuntimeError('Package '+package.name+' does not have found attribute!')
       if not package.found:
-        if self.argDB['with-'+package.package] == 1:
+        if 'with-'+package.package in self.framework.clArgDB and self.framework.clArgDB['with-'+package.package] == 1:
           raise RuntimeError('Package '+package.PACKAGE+' needed by '+self.name+' failed to configure.\nMail configure.log to petsc-maint@mcs.anl.gov.')
 
     dpkgs = Package.sortPackageDependencies(self)
